test/remoting.integration.js

```diff
@@ -238,5 +185,185 @@ describe('remoting - integration', function() {
         ];
         expect(methods).to.include.members(expectedMethods);
       });
+
+    it('has expected remote methods with model.settings.replaceOnPUT set to true', function() {
+      var storeClass = findClass('store');
+      var methods = getMethods(storeClass.methods);
+
+      // TODO: there is a bug in `strong-remoting which does not support multiple
+      // http-methods and paths; please see:
+      // `https://github.com/strongloop/strong-remoting/blob/ac3093dcfbb787977ca0229b0f672703859e52e1/lib/rest-adapter.js#L622-L631
+      var expectedMethods = [
+        'create(data:object):store POST /stores',
+        'upsert(data:object):store PATCH /stores',
+        'replaceOrCreate(data:object):store PUT /stores',
+        'replaceOrCreate(data:object):store POST /stores/replaceOrCreate',
+        'exists(id:any):boolean GET /stores/:id/exists',
+        'findById(id:any,filter:object):store GET /stores/:id',
+        'replaceById(id:any,data:object):store PUT /stores/:id',
+        'replaceById(id:any,data:object):store POST /stores/:id/replace',
+        'find(filter:object):store GET /stores',
+        'findOne(filter:object):store GET /stores/findOne',
+        'updateAll(where:object,data:object):object POST /stores/update',
+        'deleteById(id:any):object DELETE /stores/:id',
+        'count(where:object):number GET /stores/count',
+        'prototype.updateAttributes(data:object):store PATCH /stores/:id',
+        'createChangeStream(options:object):ReadableStream POST /stores/change-stream',
```

Comment:
We are duplicating many methods that are not relevant to this test. I am proposing to trim down the list of methods to only the entries for the four methods we are interested in: replaceById, replaceOrCreate, upsert, prototype.updateAttributes.